File: javascript/src/model/RingOutStatusInfo.js
(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['../ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.RingCentralApi) {
      root.RingCentralApi = {};
    }
    root.RingCentralApi.RingOutStatusInfo = factory(root.RingCentralApi.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The RingOutStatusInfo model module.
   * @module model/RingOutStatusInfo
   * @version v1.0
   */

  /**
   * Constructs a new <code>RingOutStatusInfo</code>.
   * @alias module:model/RingOutStatusInfo
   * @class
   */
  var exports = function() {




  };

  /**
   * Constructs a <code>RingOutStatusInfo</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/RingOutStatusInfo} obj Optional instance to populate.
   * @return {module:model/RingOutStatusInfo} The populated <code>RingOutStatusInfo</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) { 
      obj = obj || new exports();

      if (data.hasOwnProperty('callStatus')) {
        obj['callStatus'] = ApiClient.convertToType(data['callStatus'], 'String');
      }
      if (data.hasOwnProperty('callerStatus')) {
        obj['callerStatus'] = ApiClient.convertToType(data['callerStatus'], 'String');
      }
      if (data.hasOwnProperty('calleeStatus')) {
        obj['calleeStatus'] = ApiClient.convertToType(data['calleeStatus'], 'String');
      }
    }
    return obj;
  }


  /**
   * Status of a call
   * @member {module:model/RingOutStatusInfo.CallStatusEnum} callStatus
   */
  exports.prototype['callStatus'] = undefined;

  /**
   * Status of a calling party
   * @member {module:model/RingOutStatusInfo.CallerStatusEnum} callerStatus
   */
  exports.prototype['callerStatus'] = undefined;

  /**
   * Status of a called party
   * @member {module:model/RingOutStatusInfo.CalleeStatusEnum} calleeStatus
   */
  exports.prototype['calleeStatus'] = undefined;


  /**
   * Allowed values for the <code>callStatus</code> property.
   * @enum {String}
   * @readonly
   */
  exports.CallStatusEnum = { 
    /**
     * value: Invalid
     * @const
     */
    INVALID: "Invalid",
    
    /**
     * value: Success
     * @const
     */
    SUCCESS: "Success",
    
    /**
     * value: InProgress
     * @const
     */
    INPROGRESS: "InProgress",
    
    /**
     * value: Busy
     * @const
     */
    BUSY: "Busy",
    
    /**
     * value: NoAnswer
     * @const
     */
    NOANSWER: "NoAnswer",
    
    /**
     * value: Rejected
     * @const
     */
    REJECTED: "Rejected",
    
    /**
     * value: GenericError
     * @const
     */
    GENERICERROR: "GenericError",
    
    /**
     * value: Finished
     * @const
     */
    FINISHED: "Finished",
    
    /**
     * value: InternationalDisabled
     * @const
     */
    INTERNATIONALDISABLED: "InternationalDisabled",
    
    /**
     * value: DestinationBlocked
     * @const
     */
    DESTINATIONBLOCKED: "DestinationBlocked",
    
    /**
     * value: NotEnoughFunds
     * @const
     */
    NOTENOUGHFUNDS: "NotEnoughFunds",
    
    /**
     * value: NoSuchUser
     * @const
     */
    NOSUCHUSER: "NoSuchUser"
  };  /**
   * Allowed values for the <code>callerStatus</code> property.
   * @enum {String}
   * @readonly
   */
  exports.CallerStatusEnum = { 
    /**
     * value: Invalid
     * @const
     */
    INVALID: "Invalid",
    
    /**
     * value: Success
     * @const
     */
    SUCCESS: "Success",
    
    /**
     * value: InProgress
     * @const
     */
    INPROGRESS: "InProgress",
    
    /**
     * value: Busy
     * @const
     */
    BUSY: "Busy",
    
    /**
     * value: NoAnswer
     * @const
     */
    NOANSWER: "NoAnswer",
    
    /**
     * value: Rejected
     * @const
     */
    REJECTED: "Rejected",
    
    /**
     * value: GenericError
     * @const
     */
    GENERICERROR: "GenericError",
    
    /**
     * value: Finished
     * @const
     */
    FINISHED: "Finished",
    
    /**
     * value: InternationalDisabled
     * @const
     */
    INTERNATIONALDISABLED: "InternationalDisabled",
    
    /**
     * value: DestinationBlocked
     * @const
     */
    DESTINATIONBLOCKED: "DestinationBlocked",
    
    /**
     * value: NotEnoughFunds
     * @const
     */
    NOTENOUGHFUNDS: "NotEnoughFunds",
    
    /**
     * value: NoSuchUser
     * @const
     */
    NOSUCHUSER: "NoSuchUser"
  };  /**
   * Allowed values for the <code>calleeStatus</code> property.
   * @enum {String}
   * @readonly
   */
  exports.CalleeStatusEnum = { 
    /**
     * value: Invalid
     * @const
     */
    INVALID: "Invalid",
    
    /**
     * value: Success
     * @const
     */
    SUCCESS: "Success",
    
    /**
     * value: InProgress
     * @const
     */
    INPROGRESS: "InProgress",
    
    /**
     * value: Busy
     * @const
     */
    BUSY: "Busy",
    
    /**
     * value: NoAnswer
     * @const
     */
    NOANSWER: "NoAnswer",
    
    /**
     * value: Rejected
     * @const
     */
    REJECTED: "Rejected",
    
    /**
     * value: GenericError
     * @const
     */
    GENERICERROR: "GenericError",
    
    /**
     * value: Finished
     * @const
     */
    FINISHED: "Finished",
    
    /**
     * value: InternationalDisabled
     * @const
     */
    INTERNATIONALDISABLED: "InternationalDisabled",
    
    /**
     * value: DestinationBlocked
     * @const
     */
    DESTINATIONBLOCKED: "DestinationBlocked",
    
    /**
     * value: NotEnoughFunds
     * @const
     */
    NOTENOUGHFUNDS: "NotEnoughFunds",
    
    /**
     * value: NoSuchUser
     * @const
     */
    NOSUCHUSER: "NoSuchUser"
  };

  return exports;
}));

File: csharp/src/main/csharp/IO/Swagger/Model/RingOutStatusInfo.cs
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace IO.Swagger.Model
{
    /// <summary>
    /// 
    /// </summary>
    [DataContract]
    public partial class RingOutStatusInfo :  IEquatable<RingOutStatusInfo>
    { 
    
        /// <summary>
        /// Status of a call
        /// </summary>
        /// <value>Status of a call</value>
        [JsonConverter(typeof(StringEnumConverter))]
        public enum CallStatusEnum {
            
            [EnumMember(Value = "Invalid")]
            Invalid,
            
            [EnumMember(Value = "Success")]
            Success,
            
            [EnumMember(Value = "InProgress")]
            Inprogress,
            
            [EnumMember(Value = "Busy")]
            Busy,
            
            [EnumMember(Value = "NoAnswer")]
            Noanswer,
            
            [EnumMember(Value = "Rejected")]
            Rejected,
            
            [EnumMember(Value = "GenericError")]
            Genericerror,
            
            [EnumMember(Value = "Finished")]
            Finished,
            
            [EnumMember(Value = "InternationalDisabled")]
            Internationaldisabled,
            
            [EnumMember(Value = "DestinationBlocked")]
            Destinationblocked,
            
            [EnumMember(Value = "NotEnoughFunds")]
            Notenoughfunds,
            
            [EnumMember(Value = "NoSuchUser")]
            Nosuchuser
        }

    
        /// <summary>
        /// Status of a calling party
        /// </summary>
        /// <value>Status of a calling party</value>
        [JsonConverter(typeof(StringEnumConverter))]
        public enum CallerStatusEnum {
            
            [EnumMember(Value = "Invalid")]
            Invalid,
            
            [EnumMember(Value = "Success")]
            Success,
            
            [EnumMember(Value = "InProgress")]
            Inprogress,
            
            [EnumMember(Value = "Busy")]
            Busy,
            
            [EnumMember(Value = "NoAnswer")]
            Noanswer,
            
            [EnumMember(Value = "Rejected")]
            Rejected,
            
            [EnumMember(Value = "GenericError")]
            Genericerror,
            
            [EnumMember(Value = "Finished")]
            Finished,
            
            [EnumMember(Value = "InternationalDisabled")]
            Internationaldisabled,
            
            [EnumMember(Value = "DestinationBlocked")]
            Destinationblocked,
            
            [EnumMember(Value = "NotEnoughFunds")]
            Notenoughfunds,
            
            [EnumMember(Value = "NoSuchUser")]
            Nosuchuser
        }

    
        /// <summary>
        /// Status of a called party
        /// </summary>
        /// <value>Status of a called party</value>
        [JsonConverter(typeof(StringEnumConverter))]
        public enum CalleeStatusEnum {
            
            [EnumMember(Value = "Invalid")]
            Invalid,
            
            [EnumMember(Value = "Success")]
            Success,
            
            [EnumMember(Value = "InProgress")]
            Inprogress,
            
            [EnumMember(Value = "Busy")]
            Busy,
            
            [EnumMember(Value = "NoAnswer")]
            Noanswer,
            
            [EnumMember(Value = "Rejected")]
            Rejected,
            
            [EnumMember(Value = "GenericError")]
            Genericerror,
            
            [EnumMember(Value = "Finished")]
            Finished,
            
            [EnumMember(Value = "InternationalDisabled")]
            Internationaldisabled,
            
            [EnumMember(Value = "DestinationBlocked")]
            Destinationblocked,
            
            [EnumMember(Value = "NotEnoughFunds")]
            Notenoughfunds,
            
            [EnumMember(Value = "NoSuchUser")]
            Nosuchuser
        }

    
        /// <summary>
        /// Status of a call
        /// </summary>
        /// <value>Status of a call</value>
        [DataMember(Name="callStatus", EmitDefaultValue=false)]
        public CallStatusEnum? CallStatus { get; set; }
    
        /// <summary>
        /// Status of a calling party
        /// </summary>
        /// <value>Status of a calling party</value>
        [DataMember(Name="callerStatus", EmitDefaultValue=false)]
        public CallerStatusEnum? CallerStatus { get; set; }
    
        /// <summary>
        /// Status of a called party
        /// </summary>
        /// <value>Status of a called party</value>
        [DataMember(Name="calleeStatus", EmitDefaultValue=false)]
        public CalleeStatusEnum? CalleeStatus { get; set; }
    
        /// <summary>
        /// Initializes a new instance of the <see cref="RingOutStatusInfo" /> class.
        /// Initializes a new instance of the <see cref="RingOutStatusInfo" />class.
        /// </summary>
        /// <param name="CallStatus">Status of a call.</param>
        /// <param name="CallerStatus">Status of a calling party.</param>
        /// <param name="CalleeStatus">Status of a called party.</param>

        public RingOutStatusInfo(CallStatusEnum? CallStatus = null, CallerStatusEnum? CallerStatus = null, CalleeStatusEnum? CalleeStatus = null)
        {
            this.CallStatus = CallStatus;
            this.CallerStatus = CallerStatus;
            this.CalleeStatus = CalleeStatus;
            
        }
        
    
        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class RingOutStatusInfo {\n");
            sb.Append("  CallStatus: ").Append(CallStatus).Append("\n");
            sb.Append("  CallerStatus: ").Append(CallerStatus).Append("\n");
            sb.Append("  CalleeStatus: ").Append(CalleeStatus).Append("\n");
            
            sb.Append("}\n");
            return sb.ToString();
        }
  
        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="obj">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object obj)
        {
            // credit: http://stackoverflow.com/a/10454552/677735
            return this.Equals(obj as RingOutStatusInfo);
        }

        /// <summary>
        /// Returns true if RingOutStatusInfo instances are equal
        /// </summary>
        /// <param name="other">Instance of RingOutStatusInfo to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(RingOutStatusInfo other)
        {
            // credit: http://stackoverflow.com/a/10454552/677735
            if (other == null)
                return false;

            return 
                (
                    this.CallStatus == other.CallStatus ||
                    this.CallStatus != null &&
                    this.CallStatus.Equals(other.CallStatus)
                ) && 
                (
                    this.CallerStatus == other.CallerStatus ||
                    this.CallerStatus != null &&
                    this.CallerStatus.Equals(other.CallerStatus)
                ) && 
                (
                    this.CalleeStatus == other.CalleeStatus ||
                    this.CalleeStatus != null &&
                    this.CalleeStatus.Equals(other.CalleeStatus)
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            // credit: http://stackoverflow.com/a/263416/677735
            unchecked // Overflow is fine, just wrap
            {
                int hash = 41;
                // Suitable nullity checks etc, of course :)
                
                if (this.CallStatus != null)
                    hash = hash * 59 + this.CallStatus.GetHashCode();
                
                if (this.CallerStatus != null)
                    hash = hash * 59 + this.CallerStatus.GetHashCode();
                
                if (this.CalleeStatus != null)
                    hash = hash * 59 + this.CalleeStatus.GetHashCode();
                
                return hash;
            }
        }

    }
}
